pkg/services/govmomi/service.go

```diff
@@ -250,13 +263,179 @@ func (vms *VMService) reconcileNetworkStatus(ctx *virtualMachineContext) error {
 	return nil
 }
 
+func (vms *VMService) reconcileIPAddressClaims(ctx *virtualMachineContext) (bool, error) {
+	for devIdx, device := range ctx.VSphereVM.Spec.Network.Devices {
+		for poolRefIdx, poolRef := range device.AddressesFromPools {
+			// check if claim exists
+			ipAddrClaim := &ipamv1.IPAddressClaim{}
+			ipAddrClaimName := ipAddressClaimName(ctx.VSphereVM.Name, devIdx, poolRefIdx)
+			ipAddrClaimKey := apitypes.NamespacedName{
+				Namespace: ctx.VSphereVM.Namespace,
+				Name:      ipAddrClaimName,
+			}
+			var err error
+			if err = ctx.Client.Get(ctx, ipAddrClaimKey, ipAddrClaim); err != nil && !apierrors.IsNotFound(err) {
+				return false, err
+			}
+			if err == nil {
+				ctx.Logger.V(5).Info("IPAddressClaim found", "name", ipAddrClaimName)
+			}
+			if apierrors.IsNotFound(err) {
+				ctx.Logger.Info("creating IPAddressClaim", "name", ipAddrClaimName)
+				claim := &ipamv1.IPAddressClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      ipAddrClaimName,
+						Namespace: ctx.VSphereVM.Namespace,
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: ctx.VSphereVM.APIVersion,
+								Kind:       ctx.VSphereVM.Kind,
+								Name:       ctx.VSphereVM.Name,
+								UID:        ctx.VSphereVM.UID,
+							},
+						},
+						Finalizers: []string{infrav1.IPAddressClaimFinalizer},
+					},
+					Spec: ipamv1.IPAddressClaimSpec{PoolRef: poolRef},
+				}
+				if err = ctx.Client.Create(ctx, claim); err != nil {
+					return false, err
+				}
+				msg := "Waiting for IPAddressClaim to have an IPAddress bound"
+				markIPAddressClaimedConditionWaitingForClaimAddress(ctx.VSphereVM, msg)
+			}
+		}
+	}
+
+	return true, nil
+}
+
+func (vms *VMService) reconcileIPAddresses(ctx *virtualMachineContext) (bool, error) {
```

Comment:
nit: add a comment stating the intent of the method

Reply:
updated